packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerHelpers.ts

```diff
@@ -0,0 +1,156 @@
+            .replace(/[^a-zA-Z0-9]/gi, '');
+          const [shortId] = uuid().split('-');
+          return `${appName}${shortId}`;
+        },
+    }];
+}
+
+export function layerPermissionsQuestion(params: Permissions[]) {
+    return [
+      {
+        type: 'checkbox',
+        name: 'layerPermissions',
+        message: 'Who should have permission to use this layer?',
+        choices: [
+          {
+            name: 'Only the current AWS account',
+            value: Permissions.private,
+            checked: _.includes(params, Permissions.private),
+          },
+          {
+            name: 'Specific AWS accounts',
+            value: Permissions.awsAccounts,
+            checked: _.includes(params, Permissions.awsAccounts),
+          },
+          {
+            name: 'Specific AWS organization',
+            value: Permissions.awsOrg,
+            checked: _.includes(params, Permissions.awsOrg),
+          },
+          {
+            name: 'Public (everyone on AWS can use this layer)',
+            value: Permissions.public,
+            checked: _.includes(params, Permissions.public),
+          },
+        ],
+        default: Permissions.private,
+    }];
+}
+
+export function layerAccountAccessQuestion(defaultaccounts? : string[]) {
+    return [
+      {
+        type: 'input',
+        name: 'authorizedAccountIds',
+        message: 'Provide a list of comma-separated AWS account IDs:',
+        validate: input => {
+          const accounts = input.split(',');
+          const set = new Set();
+          for (let accountID of accounts) {
+            accountID = accountID.trim();
+            if (!/^[0-9]{12}$/.test(accountID)) {
+              return `AWS account IDs must be 12 digits long. ${accountID} did not match the criteria.`;
+            }
+            if (set.has(accountID)) {
+              return `Duplicate ID detected: ${accountID}`;
+            }
+            set.add(accountID);
+          }
+          return true;
+        },
+        default : defaultaccounts.toString()
+    }];
+}
+
+export function layerOrgAccessQuestion(defaultorgs? : string[]) {
+    return [
+      {
+        type: 'input',
+        name: 'authorizedOrgId',
+        message: 'Provide a list of comma-separated AWS organization IDs:',
+        validate: input => {
+          const orgIds = input.split(',');
+          const set = new Set();
+          orgIds.forEach(orgId => {
+            orgId = orgId.trim();
+            if (!/^o-[a-zA-Z0-9]{10,32}$/.test(orgId)) {
+              return 'The organization ID starts with "o-" followed by a 10-32 character-long alphanumeric string.';
+            }
+            if (set.has(orgId)) {
+              return `Duplicate ID detected: ${orgId}`;
+            }
+            set.add(orgId);
+          })
+          return true;
+        },
+        default : defaultorgs.toString()
+    }];
+}
+
+export function createVersionsMap(parameters : Partial<LayerParameters>,version : string){
```

Comment:
this method only needs parameters.layerPermissions

Also, why are we representing the permissions in 2 different ways? Seems like we should refactor so a conversion like this isn't necessary